supply/lib/supply/setup.rb

```diff
@@ -43,45 +42,46 @@ def store_metadata(listing)
     def download_images(listing)
       UI.message("🖼️  Downloading images (#{listing.language})")
 
-      # We cannot download existing screenshots as they are compressed
-      # But we can at least download the images
       require 'net/http'
 
-      IMAGES_TYPES.each do |image_type|
-        if ['featureGraphic'].include?(image_type)
-          # we don't get all files in full resolution :(
-          UI.message("📵  Due to a limitation of the Google Play API, there is no way for `supply` to download your existing feature graphic. Please copy your feature graphic to `metadata/android/#{listing.language}/images/featureGraphic.png`")
-          next
-        end
+      allowed_imagetypes = [Supply::IMAGES_TYPES, Supply::SCREENSHOT_TYPES].flatten
 
+      allowed_imagetypes.each do |image_type|
         begin
           UI.message("Downloading `#{image_type}` for #{listing.language}...")
 
-          url = client.fetch_images(image_type: image_type, language: listing.language).last
-          next unless url
+          urls = client.fetch_images(image_type: image_type, language: listing.language)
+          next if urls.nil? || urls.empty?
+
+          image_counter = 1 # Used to prefix the downloaded files, so order is preserved.
+          urls.each do |url|
+            url = url.gsub(url.match("=.*").to_s, "") # Remove everything after '=' (if present)
+            url = url.gsub("-rw", "") # Remove "-rw" (if present) so the url will become either 'jpg' or 'png' (https://github.com/fastlane/fastlane/pull/14322#issuecomment-473017130)
+            url = "#{url}=s0" # '=s0' param ensures full image size is returned (https://github.com/fastlane/fastlane/pull/14322#issuecomment-473012462)
```

Comment:
Can we add a verbose log output here that mentions that we messed with `url` here (if we actually did) so if Google every decides to break these URLs we have a way for users to actually find out that our messing with the `url` is the culprit here?

Reply:
Added some verbose logging. Let me know if you'd like to change it to something else, please!

https://github.com/fastlane/fastlane/pull/14322/commits/f720f66544def11d165bd8d02fbef66a8f73a429

Reply:
logging of the removal should probably only happen if something was removed, and also include the before/after of the url. that we do add `=s0` doesn't have to be logged I guess if we see the end result.

Reply:
Made these changes - https://github.com/fastlane/fastlane/pull/14322/commits/5e8c57f62c3f87c71002cc0b7a49a93351e20d7c